pkg/kubernetes/events/event_pubsub.go

```diff
@@ -36,6 +36,33 @@ func (c *osmPubsub) Publish(message PubSubMessage) {
 	c.pSub.Pub(message, message.AnnouncementType.String())
 }
 
+// Close is the Close implementation for PubSub.
+// It is synchronized, upon exit the channel is guaranteed to be both
+// unsubbed to all topics and closed.
+// This is a necessary step to guarantee garbage collection
+func (c *osmPubsub) Close(unsubChan chan interface{}) {
+	// implementation has several requirements (including different goroutine context)
+	// https://github.com/cskr/pubsub/blob/master/pubsub.go#L102
+
+	syncCh := make(chan struct{})
+	go func() {
+		// This should close the channel on the pubsub backend
+		// https://github.com/cskr/pubsub/blob/master/pubsub.go#L264
+		c.pSub.Unsub(unsubChan)
+
+		for {
+			_, ok := <-unsubChan
+			if !ok {
+				// Channel closed from pubsub
+				break
+			}
+		}
+		syncCh <- struct{}{}
+	}()
+
+	<-syncCh
```

Comment:
Since the goroutine + `syncCh` is implementing a synchronous close, why can't this be done inline without a goroutine to close the `unsubChan`?

Reply:
Discussed offline, different goroutine on  required by impl of library in use